### PR TITLE
[rllib] improve custom env docs

### PR DESCRIPTION
**Pull request by ericl**
https://github.com/ray-project/ray/issues/1440

### CODE REVIEW

File: doc/source/rllib.rst [hunk 1]
```diff
@@ -171,15 +171,18 @@ Custom Environments
 To train against a custom environment, i.e. one not in the gym catalog, you
 can register a function that creates the env to refer to it by name. The contents of the
 ``env_config`` agent config field will be passed to that function to allow the
-environment to be configured. For example:
+environment to be configured. The return type should be an `OpenAI gym.Env <https://github.com/openai/gym/blob/master/gym/core.py>`__. For example:
```
Comment (robertnishihara) [inline doc/source/rllib.rst]: When I view the rendered HTML, this link somehow gets abbreviated to https://github.com/openai/gym.

File: doc/source/rllib.rst [hunk 2]
```diff
@@ -171,15 +171,18 @@ Custom Environments
...
 
 ::
 
     import ray
     from ray.tune.registry import register_env
     from ray.rllib import ppo
 
-    env_creator = lambda env_config: MyCustomEnv(env_config)
+    def env_creator(env_config):
```
Comment (robertnishihara) [inline doc/source/rllib.rst]: for this code block (and any other python clodeblocks), can you replace `::` with `.. code-block:: python` to get the syntax highlighting? For code intended to run on the command line, we can use `.. code-block:: bash`.

File: doc/source/rllib.rst (hunk 1 above)
Comment (ericl) [reply doc/source/rllib.rst]: Weird, it looks ok to me though.

File: doc/source/rllib.rst (hunk 2 above)
Comment (ericl) [reply doc/source/rllib.rst]: Done

File: doc/source/rllib.rst (hunk 1 above)
Comment (richardliaw) [reply doc/source/rllib.rst]: This is fine for both the github rendering and on my local machine; there is another link for `OpenAI Gym MDP`, so maybe you clicked on that instead?

**Review by richardliaw — APPROVED** [review]
@robertnishihara I'm going to approve for now and if there's other issues we can always fix.

**Pull request merged**